src/Feeds.js

```diff
@@ -1,270 +1,425 @@
    * @desc Update feed balance
    * @param {Array} updates Array of updates
-   * @param {String} updates[].user_id user id to update
-   * @param {Object} updates[].wallet_name user id to update
+   * @param {String} updates[].feed_id feed id to update
+   * FIXME
+   * @param {Object} updates[].wallet_name feed id to update
    * @param {Object} updates[].amount amount
    */
-  async updateFeedBalance (updates) {
-    let res
+  async updateFeedBalance (update) {
+    if (!this.ready) throw new Err(_err.notReady)
+
+    this.validateUpdate(update)
+
+    const existingFeed = await this.db.findByFeedId(update.feed_id)
+    if (!existingFeed) throw new Err(_err.feedNotExists)
+
     try {
-      res = await Promise.all(updates.map(async (update) => {
-        if (typeof update.wallet_name !== 'string' || typeof update.user_id !== 'string') throw new Err(_err.badUpdateParam)
-        if (Number.isNaN(+update.amount)) throw new Err(_err.badUpdateParam)
-        await this.slashtags.update(update.user_id, this._getWalletFeedKey(update.wallet_name), update.amount)
-        return true
-      }))
+      // NOTE: consider storing balance on db as well
+      for (const field of update.fields) {
+        await this.slashtags.update(update.feed_id, SlashtagsFeeds.getFileName(field), field.value)
+      }
+      return { updated: true }
     } catch (err) {
+      log.err(err)
       if (err instanceof Err) throw err
       throw new Err(_err.updateFeedFailed)
     }
-    return res
   }
 
-  async startFeedBroadcast () {
-    let feeds
-    try {
-      feeds = await this.db.getAllActiveFeeds()
-    } catch (err) {
-      log.error(err)
-      throw new Error(_err.failedGettingActiveFeeds)
-    }
-
-    let res
-    try {
-      res = await Promise.all(feeds.map((user) => {
-        return this.slashtags.feed(user.user_id, {
-          announce: true
-        })
-      }))
-    } catch (err) {
-      console.log(err)
-      throw new Error(_err.failedBroadcast)
-    }
-    return {
-      feeds_started: res.length
-    }
-  }
+  async deleteFeed (args) {
+    if (!this.ready) throw new Err(_err.notReady)
+    if (!args?.feed_id) throw new Err(_err.feedIdMissing)
+    if (typeof args.feed_id !== 'string') throw new Err(_err.feedIdNotString)
 
-  async deleteUserFeed (args) {
-    if (typeof args.user_id !== 'string') throw new Err(_err.useridNotString)
-    const userId = args.user_id
+    const feedId = args.feed_id
     try {
-      const existingUser = await this.getFeedFromDb(userId)
-      if (!existingUser) {
-        log.info(`Deleting user that does not exist: ${userId}`)
+      const existingFeed = await this.getFeedFromDb(args)
+      if (!existingFeed) {
+        log.info(`Deleting feed that does not exist: ${feedId}`)
         return { deleted: true }
       }
-      await this.db.removeUser(userId)
-      await this.slashtags.destroy(userId)
+      // XXX: this needs to be atomic to prevent discrepancy between local DB and Hyperdrive
```

Comment:
Don't worry much about this, worst case scenario you have a Hyperdrive data on disk that you would rather not have.

Storage is cheap, and as long as the `sqlite` database is the source of truth, you can collect garbage at your leisure.